stringtemplate4/src/test/java/org/jdbi/v3/sqlobject/BindListTest.java

```diff
@@ -215,4 +208,35 @@ public void testSomethingByIteratorHandleDefault() {
         @SqlQuery("select id, name from something where id in (<ids>)")
         List<Something> get(@BindList Iterator<Integer> ids);
     }
+
+    //
+
+    @Test
+    public void testConditionalRendering() {
+        handle.createUpdate("insert into something(id, name) values(4, null)").execute();
+
+        List<String> allNames = handle.createQuery("select name from something")
+            .mapTo(String.class)
+            .list();
+        assertThat(allNames).hasSize(4);
+
+        List<String> nullNames = handle.createQuery("select name from something where name is null")
+            .mapTo(String.class)
+            .list();
+        assertThat(nullNames).hasSize(1);
+
+        ConditionalDao dao = handle.attach(ConditionalDao.class);
+        List<String> names = dao.get(null);
+
+        assertThat(names)
+            .describedAs("ST did not evaluate null as truthy, query did not select by `name is null`")
+            .hasSize(4);
+    }
+
+    private interface ConditionalDao {
+        // `in (null)` doesn't work on h2
+        @SqlQuery("select name from something <if(name)> where name is <name> <endif>")
```

Comment:
Should this be `name in (<name>)`?

Reply:
Also, can we get a positive test case for this `<if()`> statement?

Reply:
The comment on it explains why the query isn't `in ()`. It doesn't really matter, it's only about what BindList with StringTemplate renders to, not the query parsing. That's also why I have 2 control queries in the test, to demonstrate the essence of it.